pkgs/development/tools/geckodriver/default.nix

```diff
@@ -1,26 +1,30 @@
 { lib
-, fetchFromGitHub
+, fetchzip
 , rustPlatform
 , stdenv
 , darwin
 }:
 
-with rustPlatform; 
-
-buildRustPackage rec {
-  version = "0.22.0";
+rustPlatform.buildRustPackage {
+  version = "0.26.0";
   pname = "geckodriver";
+  sourceRoot = "source/testing/geckodriver";
 
-  src = fetchFromGitHub {
-    owner = "mozilla";
-    repo = "geckodriver";
-    rev = "v${version}";
-    sha256 = "12m95lfqwdxs2m5kjh3yrpm9w2li5m8n3fw46a2nkxyfw6c94l4b";
-  };
+  src = let
+    # Source revisions are noted alongside the binary releases:
+    # https://github.com/mozilla/geckodriver/releases
+    rev = "e9783a644016aa9b317887076618425586730d73";
+  in (fetchzip {
+    url = "https://hg.mozilla.org/mozilla-central/archive/${rev}.zip/testing";
+    sha256 = "0m86hqyq1jrr49jkc8mnlmx4bdq281hyxhcrrzacyv20nlqwvd8v";
+  }).overrideAttrs (_: {
+    unpackCmd = "unzip $curSrc";
```

Comment:
Why is this required?

Reply:
There might be a better way to do this? This is here because [unpack phases choose the command based on file extension](https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/archivers/unzip/setup-hook.sh) and `fetchzip` as far as I know doesn't provide a way to change the filename.

Reply:
How about

```
unpackPhase = ''
  unzip $src
'';
```

With a comment about why the normal unpackPhase doesn’t work?

Reply:
If you prefer it as you have it now that’s also fine.

Reply:
I'm not sure if that will work? `fetchzip` explicitly calls `unpackFile` as part of `postFetch`, but it makes sense to add a comment regarding why it's there.

Reply:
Looks like you’re right. :(